routing/router.go

```diff
@@ -2249,3 +2249,290 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 
 	return bandwidthHints, nil
 }
+
+// runningAmounts keeps running amounts while the route is traversed.
+type runningAmounts struct {
+	// amt is the intended amount to send via the route.
+	amt lnwire.MilliSatoshi
+
+	// max is the running maximum that the route can carry.
+	max lnwire.MilliSatoshi
+}
+
+// stepBack returns a new set of running amounts that would result from
+// prepending the given channel to the route. If canIncreaseAmt is set, the
+// amount may be increased if it is too small to satisfy the channel's minimum
+// htlc amount.
+func (r *runningAmounts) stepBack(policy *channeldb.ChannelEdgePolicy,
+	capacity btcutil.Amount, localChan bool, canIncreaseAmt bool) (
+	runningAmounts, error) {
+
+	// Determine max htlc value.
+	maxHtlc := lnwire.NewMSatFromSatoshis(capacity)
+	if policy.MaxHTLC != 0 {
```

Comment:
Should use `HasMaxHtlc`?

Reply:
Changed. This is the confusion that I think we shouldn't need to have.

Reply:
Hmm, looking in the link, it also checks for `MaxHTLC == 0`

Reply:
That's because the link uses `ForwardingPolicy`, which doesn't contain information about whether the bit is set or not. I'm in favor of adding the check for the bit being set here.

Reply:
Ok, it is indeed a good idea, because this function is also used for non-local channels. Changed